docs/reference.md

```diff
@@ -0,0 +1,1237 @@
+`unbind()` method, which is passed into the promise callback.
+
+```js
+var obj = $firebaseObject(ref);
+obj.$bindTo($scope, "data").then(function(unbind) {
+  // unbind this later
+  //unbind();
+});
+```
+
+### $watch(callback, context)
+
+Registers an event listener which will be notified any time there is a change to the data. Returns
+an unregister function that, when invoked, will stop notifying the callback of changes.
+
+```js
+var obj = $firebaseObject(ref);
+var unwatch = obj.$watch(function() {
+  console.log("data changed!");
+});
+
+// at some time in the future, we can unregister using
+unwatch();
+```
+
+### $destroy()
+
+Calling this method cancels event listeners and frees memory used by this object (deletes the
+local data). Changes are no longer synchronized to or from the database.
+
+
+## $firebaseArray
+
+The `$firebaseArray` service takes an optional [`firebase.database.Reference`](https://firebase.google.com/docs/reference/js/#firebase.database.Reference) or
+[`firebase.database.Query`](https://firebase.google.com/docs/reference/js/#firebase.database.Query) and returns a JavaScript array which contains the data at the
+provided location in Firebase and some extra AngularFire-specific fields. If no `Reference` or `Query` is provided, then the root of the Firebase Database will be used. Note that the data will not be available immediately since retrieving
+it is an asynchronous operation. You can use the `$loaded()` promise to get notified when the data
+has loaded.
+
+This service automatically keeps this local array in sync with any changes made to the remote
+database. This is a **PSEUDO READ-ONLY ARRAY** suitable for use in directives like `ng-repeat`
+and with Angular filters (which expect an array).
+
+While using read attributes and methods like `length` and `toString()` will work great on this array,
+you should avoid directly manipulating the array. Methods like `splice()`, `push()`, `pop()`,
+`shift()`, `unshift()`, and `reverse()` will cause the local data to become out of sync with the
+server. Instead, utilize the `$add()`, `$remove()`, and `$save()` methods provided by the service to
+change the structure of the array. To get the id of an item in a $firebaseArray within `ng-repeat`, call `$id` on that item.
+
+``` js
+// JavaScript
+app.controller("MyCtrl", ["$scope", "$firebaseArray",
+  function($scope, $firebaseArray) {
+    var ref = firebase.database().ref();
+    var list = $firebaseArray(ref);
+
+    // add an item
+    list.$add({ foo: "bar" }).then(...);
+
+    // remove an item
+    list.$remove(2).then(...);
+
+    // make the list available in the DOM
+    $scope.list = list;
+  }
+]);
+```
+
+``` html
+<!-- HTML -->
+<li ng-repeat="item in list | filter:name">{{ item | json }}</li>
+```
+
+The `$firebaseArray` service can also take a
+[query](https://firebase-dot-devsite.googleplex.com/docs/database/web/retrieve-data) to only sync
```

Comment:
I think this is a staging URL? Should be https://firebase.google.com/docs/database/web/retrieve-data?